### PR TITLE
Add pre-push hook check to block leftover .only()s

### PR DESCRIPTION
**Pull request by charisk**
## Checklist
N/A:

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/scripts/forbid-mocha-only [hunk 1]
```diff
@@ -0,0 +1,7 @@
+if grep -rq --include '*.test.ts' 'it.only\|describe.only' ..; then
```
[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/scripts/forbid-mocha-only]: I just realised that all the tests are actually in ./test, so I've updated the command to just look in there.

File: extensions/ql-vscode/scripts/forbid-mocha-only [hunk 2]
```diff
@@ -0,0 +1,6 @@
+if grep -rq --include '*.test.ts' 'it.only\|describe.only' './test'; then
```
Comment (shati-patel) [inline extensions/ql-vscode/scripts/forbid-mocha-only:1]: > I just realised that all the tests are actually in ./test, so I've updated the command to just look in there.

⚠️ 
The `./test` folder doesn't contain all tests! (I think only the "pure" tests that don't rely on any VS Code APIs.)
The other tests are in `./src/vscode-tests` 🧪

Comment (charisk) [reply extensions/ql-vscode/scripts/forbid-mocha-only]: 🤦‍♀️ thanks, I missed those. Do you happen to know if there is a reason those tests are there?

I've updated to look in ./test and ./src.

Comment (shati-patel) [reply extensions/ql-vscode/scripts/forbid-mocha-only]: > Do you happen to know if there is a reason those tests are there and not in ./test?

I asked the same question when I first started looking at the tests, but I can't quite remember the answer 😳 
The tests in `./src` are integration tests and need access to import APIs and load up the extension, so maybe they need to be next to the actual source code? 🙈

Comment (charisk) [reply extensions/ql-vscode/scripts/forbid-mocha-only]: Ok thanks, was just being nosey :D

**Review by shati-patel — APPROVED** [review]

File: extensions/ql-vscode/scripts/forbid-mocha-only (hunk 2 above)
Comment (aeisenberg) [reply extensions/ql-vscode/scripts/forbid-mocha-only]: @shati-patel is exactly right. There's no _good_ reason for the separation other than that they are run differently. I don't think it's really worth the effort to combine them, but it is confusing.

**Pull request merged**